### DISCUSSION TITLE
support stand alone playback for the watch

### DISCUSSION THREAD

**Discussion by andybozy** [TortugaPower/BookPlayer#902] [General]
It is not a new idea, but there is no dedicated discussion about how to implement support for stand alone playback for the watch.
This would put this app ahead of Apple Books and Amazon Audible that both limits transfer of all books not bought in their store.

What do you think?

**Comment by andybozy** [comment]
I actually found the fork where this feature is being implemented, not a new idea I think.
The work on it started about 2019 and seems abandoned.
If I can contribute I will.
Someone else is interested?

**Comment by GianniCarlo** [comment]
I left a comment on [this ticket](https://github.com/TortugaPower/BookPlayer/issues/651#issuecomment-1537513781), but basically after v5.0.0 with sync across devices is released, next to tackle is stand alone playback for the watch, with the first iteration just direct communication from the watch to the server to either download or stream the item. After that we can look into transferring the file from the phone to the watch for playback without relying on the server

**Reply by Daskarzine** [reply]
Hi - can I check if anything came of this? Would love to be able to stream or load a file to the watch to play an audiobook without the phone - it's the one missing piece for me for this great app!

**Reply by GianniCarlo** [reply]
@Daskarzine I'm making progress, I have a rough implementation on a blank/separate project, I just have to keep adding what is essential from the main player class into the new cleaner player class on the watch, and replace the main player

**Reply by Daskarzine** [reply]
GREAT! Thanks for the reply and update!

**Reply by steveylang** [reply]
Thank you so much for working on this, I just sent a virtual tip via the iOS app as a small token of appreciation. Also- hope your young one is doing well!
